src/worker/prim-proxy.js

```diff
@@ -92,20 +117,113 @@ class PrimProxy {
         this.post(PrimProxy.opcodeMap.setY, { Y: y });
     }
 
+    // Does not work? No tests for any async functions
```

Comment:
Yes, these will have to be implemented after [this](https://linear.app/bxcoding/issue/BXC-45/scratch-variable-handling) gets solved